test/box_test.dart

```diff
@@ -168,6 +170,71 @@ void main() {
     expect(removed, equals(3));
   });
 
+  test("simple write in txn works", () {
+    int count;
+    write_func() {
+      box.putMany(simple_items);
+    }
+    store.runInTransaction(TxMode.Write, write_func);
+    count = box.count();
+    expect(count, equals(6));
+  });
+
+  test("failing transactions", () {
+    //fill box; delete outside of transaction -> then delete again in transaction -> should return false, due to ObjBoxError 404
+    int id = box.put(TestEntity.initInteger(5));
+    box.remove(id);
+    bool removed = true;
+    store.runInTransaction(TxMode.Write, () {
+      removed = box.remove(id);
+    });
+    expect(removed, equals(false));
+  });
```

Comment:
what's being tested here? AFAICT it still commits the transaction. remove doesn't throw the 404 error

Reply:
Yeah, was kinda useless; replaced it :)